src/generators/modelsGenerator.ts

```diff
@@ -332,26 +356,43 @@ function buildResponseType(
     ...(bodyProperties?.internalResponseProperties || []),
     ...(headersProperties?.internalResponseProperties || [])
   ];
-  const innerTypeWriter = Writers.objectType({
-    properties: [
-      ...(bodyProperties?.mainProperties || []),
-      {
-        name: "_response",
-        docs: ["The underlying HTTP response."],
-        type: innerResponseProperties.length
-          ? Writers.intersectionType(
-              "coreHttp.HttpResponse",
-              Writers.objectType({
-                properties: innerResponseProperties
-              })
-            )
-          : "coreHttp.HttpResponse",
-        leadingTrivia: writer => writer.blankLine()
-      }
-    ]
-  });
 
-  let intersectionTypes: WriterFunctionOrValue[] = [innerTypeWriter];
+  let intersectionTypes: WriterFunctionOrValue[] = [];
+  let innerTypeWriter: WriterFunctionOrValue = Writers.objectType({});
+
+  if (optionsBag.useCoreV2) {
+    if (
+      bodyProperties?.mainProperties &&

```

Comment:
nit: you may be able to make this in one line. `bodyProperties?.mainProperties?.length`

Reply:
Done